src/modes/draw_polygon.js

```diff
@@ -58,29 +43,41 @@ module.exports = function(ctx) {
       ctx.ui.setActiveButton(Constants.types.POLYGON);
       this.on('mousemove', CommonSelectors.true, handleMouseMove);
       this.on('click', CommonSelectors.true, handleClick);
-      this.on('keyup', CommonSelectors.isEscapeKey, stopDrawingAndRemove);
-      this.on('keyup', CommonSelectors.isEnterKey, finish);
+      this.on('keyup', CommonSelectors.isEscapeKey, () => {
+        ctx.store.delete([polygon.id], { silent: true });
+        ctx.events.changeMode(Constants.modes.SIMPLE_SELECT);
+      });
+      this.on('keyup', CommonSelectors.isEnterKey, () => {
+        ctx.events.changeMode('simple_select', { featureIds: [polygon.id] });
+      });
     },
 
     stop: function() {
       doubleClickZoom.enable(ctx);
       ctx.ui.setActiveButton();
 
-      // If it's invalid, just destroy the thing
-      if (!polygon.isValid()) {
+      polygon.removeCoordinate(`0.${currentVertexPosition}`);
```

Comment:
hmmm... I'm not seeing this locally.
![check](https://cloud.githubusercontent.com/assets/1551510/16432504/49820f02-3d53-11e6-9469-8b3abf20e711.gif)